cdn/main.tf

```
@@ -318,3 +323,14 @@ resource "azurerm_dns_cname_record" "cdnverify" {
 
   tags = var.tags
 }
+
+resource "azurerm_dns_cname_record" "custom_subdomain" {
+  name                = var.cname_record_name
```

```suggestion
  name                = raplace(var.hostname, var.dns_zone_name, "")
```
try something like this to remove `cname_record_name` var